app/controllers/locations_controller.rb

```diff
@@ -31,6 +31,24 @@ def external_search
     }, status: :internal_server_error
   end
 
+  def map_playground
+    # Show all viewable locations
+    field_id = MetadataField.find_by(name: "collection_location")
+    viewable = current_power.samples.pluck(:id)
```

Comment:
As discussed, we should try to stop using large lists of sample ids, in lieu of using the actual conditions to the database. In this case, something like:
```
samples_with_metadata = samples
                            .includes(metadata: :metadata_field)
                            .where(metadata: { metadata_field_id: metadata_field.metadata_field_id })
```

Reply:
👍